src/main/java/nl/hsac/fitnesse/fixture/slim/web/RichFacesTest.java

```diff
@@ -160,10 +160,10 @@ protected void waitForJsfAjaxImpl(boolean checkLocation) {
         try {
             // if jsf is present on page, add an event listener that will be triggered when next Ajax request completes
             if (checkLocation) {
-                waitForJavascriptCallback("if(!window.jsf||window.location.href!==arguments[0]){callback();}else{jsf.ajax.addOnEvent(function(data){if(data.status!='begin')callback();});}",
+                waitForJavascriptCallback("if(!window.jsf||window.location.href!==arguments[0]){callback();}else{jsf.ajax.addOnEvent(function(data){if(data.status=='success')callback();});}",
                         previousLocation);
             } else {
-                waitForJavascriptCallback("if(!window.jsf){callback();}else{jsf.ajax.addOnEvent(function(data){if(data.status!='begin')callback();});}");
+                waitForJavascriptCallback("if(!window.jsf){callback();}else{jsf.ajax.addOnEvent(function(data){if(data.status=='success')callback();});}");

```

Comment:
I explicitly waited on anything else than status 'begin' to ensure we also stop waiting when an error is returned by the ajax call. Does the current change still stop waiting on error (what other statuses exist than success and begin)?

Reply:
Agreed